### PR TITLE
Add Unit Tests for Purchases with Ramp Pricing

### PR DESCRIPTION
**Pull request by btruncali1**
Added unit tests to support Purchases with ramp priced subscriptions.

Added the following unit tests: `InvoicePurchaseWithRampPlan`, `InvoicePurchaseWithCustomRamps`, `PreviewPurchaseWithRamps`, `PendingPurchaseWithRamps`, `AuthAndCapturePurchaseWithRamps`

### CODE REVIEW

File: Test/PurchaseTest.cs [hunk 1]
```diff
@@ -29,6 +29,116 @@ public void InvoicePurchase()
             Assert.Equal(collection.ChargeInvoice.Address.Company, "Acme Software");
         }
 
+        [RecurlyFact(TestEnvironment.Type.Integration)]
+        public void InvoicePurchaseWithRampPlan()
+        {
+            var account = CreateNewAccountWithBillingInfo();
+            var currency = "USD";
```
Comment (csampson) [inline Test/PurchaseTest.cs:36]: nit: could just hardcode this

File: Test/PurchaseTest.cs [hunk 2]
```diff
@@ -29,6 +29,116 @@ public void InvoicePurchase()
...
+            PlansToDeactivateOnDispose.Add(plan);
+
+            var purchase = new Purchase(account.AccountCode, currency);
+            purchase.Account.BillingInfo = account.BillingInfo;
+
+            var sub = new Subscription(plan.PlanCode);
+            purchase.Subscriptions.Add(sub);
+
+            var collection = Purchase.Invoice(purchase);
+            int firstRampUnitAmount = plan.RampIntervals[0].Currencies[0].UnitAmountInCents;
```
Comment (csampson) [inline Test/PurchaseTest.cs:46]: For my own understanding, shouldn't `var` be fine here, since the type is easily inferred?

File: Test/PurchaseTest.cs (hunk 1 above)
Comment (csampson) [inline Test/PurchaseTest.cs]: nit: could just hardcode these currency values

File: Test/PurchaseTest.cs [hunk 3]
```diff
@@ -29,6 +29,116 @@ public void InvoicePurchase()
...
+            PlansToDeactivateOnDispose.Add(plan);
+            var account = CreateNewAccountWithBillingInfo();
+
+            var sub = new Subscription(account, plan, "USD");
+            sub.RampIntervals = GetMockSubscriptionRampIntervals(2);
+
+            var purchase = new Purchase(sub.AccountCode, currency);
+            purchase.Account.BillingInfo = sub.Account.BillingInfo;
+            // Randomly generate an email address
+            purchase.Account.Email = Guid.NewGuid().ToString().Substring(0, 8) + "@test.com";
```
Comment (csampson) [inline Test/PurchaseTest.cs:123]: nit: any reason not to just "user@example.com"?

**Review by mwlang — APPROVED** [review]
LGTM.  Some back and forth on what's what in Slack with Brett, but overall, this is pretty straightforward and cleanly implemented.

File: Test/BaseTest.cs [hunk 4]
```diff
@@ -268,6 +268,7 @@ public Subscription CreateNewRampSubscription(int numberOfRamps)
         public Plan CreateNewRampPlan(int numberOfRamps)
         {
             var plan = new Plan(GetMockPlanCode(), GetMockPlanName()) { Description = "Create Ramp Plan" };
+            plan.TaxExempt = true;
```
Comment (csampson) [inline Test/BaseTest.cs:271]: What's the reason for this change?

Comment (btruncali1) [reply Test/BaseTest.cs:271]: Because in the prod site, if i dont do this then taxes are included in the Totals coming back and therefore they dont match the first ramp value

Comment (btruncali1) [reply Test/BaseTest.cs:271]: Comes from this in the AC:

```
Associated plans have NO tax.
Associated purchases have NO gift cards or other discounts
```

File: Test/PurchaseTest.cs (hunk 3 above)
Comment (btruncali1) [reply Test/PurchaseTest.cs:123]: Just to prevent the test from being flakey.  i guess i wasnt wanting accounts to be out there all with the same email incase something somehow didnt make it to the account.close().  Everything else in the tests are setup to get new random string, for example plan_code or plan name.

File: Test/PurchaseTest.cs (hunk 2 above)
Comment (btruncali1) [reply Test/PurchaseTest.cs:46]: it could be "fine" there.  But better to be explicit when you can.

Comment (btruncali1) [reply Test/PurchaseTest.cs:46]: Typically when its the base types like `int` or `string` or `decimal`, youre explicit about it in c# when you can be

**Review by csampson — APPROVED** [review]
